codex-rs/tui/src/conversation_history_widget.rs

```diff
@@ -232,6 +237,87 @@ impl ConversationHistoryWidget {
             }
         }
     }
+
+    pub fn previous(&mut self) -> Option<String> {
```

Comment:
Perhaps we should have a separate struct that is associated with the `BottomPane` that should be managing history instead?